### PR TITLE
Atari dqn

### PR DESCRIPTION
**Pull request by lgraesser**

### CODE REVIEW

**Comment by lgraesser** [comment]
New features:
- Atari replay memory
- Stacked replay memory
- Memory profiling
       - Function timer decorator, currently muted
       - Object size profiler
- Parameterized the learning rate decay factor and fixed 
- Added last state guard to replay memories

See dqn_atari.json for an example spec with the same network and similar parameter settings from the "Human Level Control Through Deep Reinforcement Learning" paper
https://www.nature.com/articles/nature14236
      
For pyTorch and hyperparam search to play nicely, you need to set OMP_NUM_THREADS=1 at the command line as follows before running search
export OMP_NUM_THREADS=1
